eng/tools/versioning/set-dev.js

```diff
@@ -0,0 +1,68 @@
+let argv = require("yargs")
+  .options({
+    "build-id": {
+      type: "string",
+      describe: "build ID suffix to give the package (e.g. usually YYYYMMDD.r)",
+      demandOption: true
+    },
+    "repo-root": {
+      type: "string",
+      describe: "root of the repository (e.g. ../../../)",
+      demandOption: true
+    },
+    service: {
+      type: "string",
+      describe:
+        "service directory whose packages should be updated (if not set updates all directories)"
+    }
+  })
+  .help().argv;
+
+const path = require("path");
+const semver = require("semver");
+const versionUtils = require("./VersionUtils");
+
+function getDevVersion(currentVersion, buildId) {
+  const parsedVersion = semver.parse(currentVersion);
+  return `${parsedVersion.major}.${parsedVersion.minor}.${parsedVersion.patch}-dev.${buildId}`;
+}
+
+async function main(argv) {
+  const buildId = argv["build-id"];
+  const repoRoot = argv["repo-root"];
+  const service = argv["service"];
+
+  const rushSpec = await versionUtils.getRushSpec(repoRoot);
+
+  console.log(`Updating packages with Build ID ${buildId}`);
+
+  const scopedPackages = rushSpec.projects.filter(package =>
+    service ? package.projectFolder.startsWith(`sdk/${service}`) : true
```

Comment:
what happens when we pass in `*` in place of `$(service)` ?

Reply:
There is no folder named `*` under the SDK so no dev versions will set. Instead of passing `*` we should pass nothing into that argument (it is not a required argument) and it will update all. However, I think we're using globbing probably everywhere else so it might make sense to use more advanced pattern matching here.